T2B1 PIN dialogs

grdddj opened the pull request:
Fixes https://github.com/trezor/trezor-firmware/issues/3090
- improve the designs of PIN screens

Missing designs:
- [ ] remove + update PIN dialog
- [ ] wipe code screens

This time I did not record any UI diffs, so all the changes so far are visible in `differing_screens.html`

Hannsek commented:
<img width="244" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/e3a7df5d-171a-465a-89af-20bc7e1e6155">
"turn on" instead of "enable:

Please add vspace between first 2 paragraphs. 

it should look like this: 
<img width="511" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/1384b83e-bece-400d-9e37-da1634be2318">


instead of this: <img width="264" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/a946b606-e878-4f0b-868f-f2c304d747c4">


vertical space between upper border and header: 
<img width="454" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/d19e8cb6-2e9f-4c66-8c27-2f7d31b5ac9f">
Please delete it.

It seems that this copy is not centered: 
<img width="312" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/f42fdc64-dd2b-4ce2-91dd-fcad1b25d114">

"turned" on the next line please: 
<img width="310" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/8a7510db-0a61-4b35-b55b-e6b957f09163">

matejcik commented on core/embed/rust/src/ui/model_tr/layout.rs:
```diff
@@ -748,6 +748,37 @@ extern "C" fn tutorial(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_centered_popup_middle_button(
```
shouldn't this be `show_error` or whichever the equivalent on TT side is?

(in general please try to use semantic names. when we change the layout of "show error", we don't want to be renaming this to `leftaligned_popup_with_right_icon`)

grdddj replied:
Right, renamed to `show_error` in [f813823](https://github.com/trezor/trezor-firmware/pull/3096/commits/f813823b3cfaaf6c6425e630335d01f32267b808)

I am implementing very similar functionality on other current PRs, so after all is merged, we might want to do some review of the whole Rust API

grdddj commented:
@Hannsek  Done all the changes in [230f59c](https://github.com/trezor/trezor-firmware/pull/3096/commits/230f59c7e871163008e5b1f626eeb3f680b8e861)

Regarding the not-centered wipe-code warning, we do center it vertically, between the top and buttons. It does not look like centered here, because the buttons take up to 13px normally (in height), but here only 9px height is visible

grdddj commented:
> <img alt="image" width="248" src="https://user-images.githubusercontent.com/57008159/249581311-46b934b7-0b2e-466f-9cb1-2978dc6a4aa6.png"> -> <img alt="image" width="132" src="https://user-images.githubusercontent.com/57008159/249581461-d3de245f-aedc-4bbe-9b74-3fc836be6072.png">
> 
> Header should be uppercase <img alt="image" width="265" src="https://user-images.githubusercontent.com/57008159/249581814-76b3e388-08c7-46df-ad3c-7493f19b1cf4.png">
> 
> When there is more characters/numbers thank can fit on the screen, currently, there are three dots. Please remove it and let there only asterisks.

Done these in [00e7034](https://github.com/trezor/trezor-firmware/pull/3096/commits/00e703433d0542c96dc2a16c0b7991fa5086a9b2), together with removing the second set of screen (4-50 digits etc.) from wipecode setting

Hannsek commented:
The "success" screen should look like this: 
<img width="258" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/2ccfc68c-f52a-4221-bed8-46e49cb1ddbe">
not like this
<img width="249" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/c73035ae-1e76-489b-8d94-b497abfa542c">

This also applies for other "success" screen. Should I create a separate issue for that?

<img width="230" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/d4fdd58a-7186-4dbb-9aae-5efea6f1366c">
The dash ("_") or the center number is not horizontally centered.
<img width="230" alt="image" src="https://github.com/trezor/trezor-firmware/assets/57008159/eca37aaa-2ec9-4f8f-a36b-3b1ef359962a">

The new line for "wipe code" is not necessary (this also applies for model T), but I will put here the designs for wipe code later.

matejcik commented:
persistence test for TT is broken:
![image](https://github.com/trezor/trezor-firmware/assets/596073/f63107ba-cdc5-465c-9056-b4d3382fec5b)
(fwiw this image doesn't look right, there should not be a line break before "wipe code", right?

Hannsek commented:
RIght, it was like this in model R before for some reason. It is addressed in the comment above.

matejcik commented:
ACK on code side.

I'm still going to be obstinate about the "enable PIN" screens. Four is _not_ an acceptable number of screens before getting to do the thing I'm doing.

grdddj commented:
> The "success" screen should look like this: <img alt="image" width="258" src="https://user-images.githubusercontent.com/57008159/249758579-2ccfc68c-f52a-4221-bed8-46e49cb1ddbe.png"> not like this <img alt="image" width="249" src="https://user-images.githubusercontent.com/57008159/249758551-c73035ae-1e76-489b-8d94-b497abfa542c.png">
> 
> This also applies for other "success" screen. Should I create a separate issue for that?

This layout - gray colors instead of green - should be used in "success screens for turning security things off"? If so, please create a new issue, it is its own thing.

grdddj commented:
I have added the newline because of this screen, because it cannot fit one line completely:

![image](https://github.com/trezor/trezor-firmware/assets/42543243/d769853a-d8fa-409e-97fa-9ef155a37821)

Deleted the newline now

Hannsek commented:
> This layout - gray colors instead of green - should be used in "success screens for turning security things off"?

All success screens should look like this.

grdddj commented:
> 

So

> > This layout - gray colors instead of green - should be used in "success screens for turning security things off"?
> 
> All success screens should look like this.

So that really deserves a separate issue for "TT success screens"

Hannsek commented:
https://github.com/trezor/trezor-firmware/issues/3133

grdddj commented:
Merged in https://github.com/trezor/trezor-firmware/pull/3135
